regulation-worker/internal/delete/batch/batch.go

```diff
@@ -338,62 +299,44 @@ func (b *Batch) upload(_ context.Context, uploadFileAbsPath, actualFileName, abs
 	return nil
 }
 
-func (b *Batch) createPatternFile(userAttributes []model.User) (string, error) {
-	pkgLogger.Debug("creating a file with pattern to be searched & deleted.")
+type BatchManager struct {
+	FMFactory filemanager.FileManagerFactory
+}
 
-	searchObject := make([]byte, 0)
+func (*BatchManager) GetSupportedDestinations() []string {
+	return supportedDestinations
+}
 
-	for _, users := range userAttributes {
-		searchObject = append(searchObject, "/"...)
-		searchObject = append(searchObject, "\"userId\": *\""...)
-		searchObject = append(searchObject, users.ID...)
-		searchObject = append(searchObject, "\"/d;"...)
-	}
+func getLocalFileHandlers(destType string) map[string]filehandler.LocalFileHandler {
+	var gzipHandler filehandler.LocalFileHandler
 
-	PatternFilePtr, err := os.CreateTemp(b.TmpDirPath, "")
-	if err != nil {
-		pkgLogger.Errorf("error while creating pattern file: %v", err)
-		return "", fmt.Errorf("error while creating patternFile: %w", err)
+	if destType == "S3" {
+		gzipHandler = filehandler.NewGZIPLocalFileHandler(filehandler.CamelCase)
 	}
-	defer PatternFilePtr.Close()
 
-	_, err = PatternFilePtr.Write(searchObject)
-	if err != nil {
-		pkgLogger.Errorf("error while writing pattern: %v", err)
-		return "", fmt.Errorf("error while writing pattern:%w", err)
+	// S3_DATALAKE is a warehouse destination, so in order
+	// to send events into a warehouse destination, we simply snake_cased
+	// so the gziphandler needs to be created with Snakecasing `user_id` in mind
+	if destType == "S3_DATALAKE" {
+		gzipHandler = filehandler.NewGZIPLocalFileHandler(filehandler.SnakeCase)
 	}
```

Comment:
What if we introduce additional destinations, will the default path work for them or should we error ?

Reply:
I'll improve this check. I think you are right, we need to handle this correctly. But the problem is we have a check before on the destination we are handling. This check nonetheless needs to be improved.